pkg/secret/secret_test.go

```diff
@@ -0,0 +1,167 @@
+package secret
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+)
+
+type HelperKubernetes struct{}
+
+func (k *HelperKubernetes) ResourceContains(namespace, value string, resource schema.GroupVersionResource) (bool, error) {
+	if "nameA" == value {
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+var now = metav1.Now()
+var testNamespace = "testNamespace"
+
+func Test_ListNamesAndLabels(t *testing.T) {
+	tests := []struct {
+		name        string
+		secrets     []v1.Secret
+		secretNames []string
+		labels      []string
+		err         error
+	}{
+		{
+			name:        "ShouldParseNamesAndLabels",
```

Comment:
Better naming convention: `Given<Precondition>_When<Condition>_Then<Result>`, e.g. `GivenListOfSecrets_WhenLabelsDefined_ThenReturnNameAndLabels`

That being said, there's no test case that covers a case when there are no labels set on a secret. Can you add one?